### PR TITLE
[P3 Test][Team 6]Add Task3: Compare Index with Compression

### PR DESCRIPTION
**Pull request by zhongwew**
- What's your team number?
    Team 6

- What is the functionality being tested?
    The effect of Delta-compression in building positional index
    Method:
        compare the IO counters between non-compression and compression

- Describe your tests briefly:
    To compare the difference, we need to set up two different index managers.
    In test1:
        We test while writing documents to disk, the difference between IO counters
    In test2:
        We test while searching for phrase in document, the difference between IO counters

- Does each test case have comments/documentation?
    Yes

### CODE REVIEW

**Comment by chenghz4** [comment]
nice work! 
However, since every single test finishes, the after function will execute and delete all the files. For test 2, it wont get any result for the search because what test1 inserted is gone before test 2 start.

Also it would have been nicer if you had checked the compression of inverted list and positional list separately, it would be better.
For example for test1, insert 1000 documents with the same short text for testing the compression of inverted list.
test2, you can insert the documents with really long text to test the positional list

File: src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+    private InvertedIndexManager createPositionalManager(String path, Analyzer analyzer, boolean ifCompress){
+        File directory = new File(path);
+        if (!directory.exists()) {
+            directory.mkdirs();
+        }
+        if(ifCompress)
+            return new InvertedIndexManager.createOrOpenPositional(path, analyzer, new DeltaVarLenCompressor());
+        else
+            return new InvertedIndexManager.createOrOpenPositional(path, analyzer, new NaiveCompressor());
```
Comment (shaheming) [inline src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java]: You should delete new there, can not compile

File: src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            nonCompressManager.searchPhraseQuery(keywords);
+        nonCompressReadCounter = PageFileChannel.readCounter;
+        PageFileChannel.resetCounters();
+
+        for(int i = 0; i<10; i++)
+            compressManager.searchPhraseQuery(keywords);
+        compressReadCounter = PageFileChannel.readCounter;
+        PageFileChannel.resetCounters();
+
+        assertEquals(true, compressReadCounter/nonCompressReadCounter < 2/3);
```
Comment (shaheming) [inline src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java]: 2 / 3 == 0

File: src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        PageFileChannel.resetCounters();
+
+        //Compress manager add documents
+        for(int i = 0; i<100; i++)
+            compressManager.addDocument(sampleDoc);
+        compressManager.flush();
+        compressWriteCounter = PageFileChannel.writeCounter;
+        PageFileChannel.resetCounters();
+
+        assertEquals(true, compressWriteCounter/nonCompressWriteCounter < 2/3);
```
Comment (shaheming) [inline src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java]: 2/3 == 0

**Comment by Haixiang6123** [comment]
Good job. I think you can add a test case for testing empty text, empty document. Also, like shaheming said, 2/3 equals 0. And you need to make them in double or float type first and then do the division.

File: src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java (hunk 3 above)
Comment (zhongwew) [reply src/test/java/edu/uci/ics/cs221/Team6IndexCompressionTest.java]: Fixed. Thanks for pointing that out
